iterators_task_1/student/src/impl.cpp

```diff
@@ -1,2 +1,39 @@
 #include "impl.h"
+#include <vector>
+#include <algorithm>
 
+std::string reverseWords(const std::string& input){
+    std::stringstream sstream(input);
+    std::vector<TokenString> tokens {std::istream_iterator<TokenString>{sstream}, std::istream_iterator<TokenString>{}};
+
+
+    tokens.erase(std::remove_if(tokens.begin(), tokens.end(), [](TokenString token){
+        return token.isEmpty();
+    }), tokens.end());
+
+    std::vector<std::string> reversedTokens;
+    std::transform(tokens.rbegin(), tokens.rend(), std::back_inserter(reversedTokens), [](const TokenString& token){
+        return token.getToken();
+    });
+
+    sstream = std::stringstream();
```

Comment:
please don't:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es26-dont-use-a-variable-for-two-unrelated-purposes